proto/acl.proto

```diff
@@ -0,0 +1,16 @@
+syntax = "proto3";
+
+package acl;
+
+// Access control list for BuildBuddy objects.
+message ACL {
+  Permissions owner_permissions = 1;
+  Permissions group_permissions = 2;
+  Permissions others_permissions = 3;
+
```

Comment:
maybe tyler has ideas here - but how do we know whether the currently authenticated user is an owner, in group, or an other?

Reply:
Can we add "user_id" and "group_id" fields to the outer level ACL proto? This nicely mirrors unix perms and also lets us check fairly easily by comparing:

user.id == acl.user_id && user.read OR
user.selectedGroup.id == acl.group_id && group.write

Reply:
Done